lib/model/model.go

```diff
@@ -809,7 +809,6 @@ func (m *model) NeedSize(folder string) db.Counts {
 			return true
 		})
 	}
-	result.Bytes -= m.progressEmitter.BytesCompleted(folder)
```

Comment:
Why is this going away?

Reply:
This method is only used in folder summaries, where it is problematic (see https://github.com/syncthing/syncthing/pull/5815#issuecomment-505416623) and per its docstring:

> // NeedSize returns the number and total size of currently needed files.

Reply:
I think this should stay, as it changes what the API returns for no good reason.

Reply:
> no good reason

The reason is to fix #5811 as I explained - I'd expect a bit more of an argument to judge that "no good reason".

The other option I see would be to add another method to the model API: `DownloadedBytes`, which cold then be added to folder summaries, which could then be used in the UI to do the accounting - but that's what I call adding an API method for no good reason...

I will happily consider concrete suggestions.

Reply:
Right, I was under the assumption this number was not used in the UI, so you are removing this to avoid double deductions?
What happens when a file download is finished, are we guaranteed a new summary event? 
Because files disappear from scope.progress, and percentage drops back down?

Reply:
I don’t think there are any guarantees, and there are timing aspects. There will likely be a period of time between the download progress deregistering and a folder update. I suggest we accept that the percentage drops back momentarily, or perhaps alter the GUI so that only folder update messages are allowed to reduce the percentage. The latter might be complicated...

Reply:
I didn't think of the progress being dropped, thanks for pointing it out. Dropping percentages is ugly indeed, I'd go so far to say that I'd rather not reflect partial file downloads in the progress (like it is today) than having that. Weirdly I tested quite a lot manually and never saw this happening (I saw it stay at 100% for some time rather often though). I'll have a look whether I can get it to work without timing problems (or show that it already works :) ).